tensorflow/core/kernels/mkl/mkl_fused_batch_norm_op_test.cc

```diff
@@ -222,6 +312,133 @@ class FusedBatchNormOpTest : public OpsTestBase {
+              {}, {"x_backprop", "scale_backprop", "offset_backprop"}, {},
+              &output_tensors));
+
+          *x_backprop_tensor = output_tensors[0];
+          *scale_backprop_tensor = output_tensors[1];
+          *offset_backprop_tensor = output_tensors[2];
+        };
+
+    const GraphRunnerGrad run_mkl =
+        [this](const Tensor& input, const Tensor& filter,
+               const Tensor& y_backprop, const Tensor& scale,
+               const Tensor& mean, const Tensor& variance,
+               const Tensor& res_sp3, Tensor* x_backprop_tensor,
+               Tensor* scale_backprop_tensor, Tensor* offset_backprop_tensor,
+               const float epsilon) {
+          Tensor conv2d_output, conv2d_meta_output;
+          Conv2DOpTest<T> conv2d_test;
+          conv2d_test.RunConv2D(input, filter, &conv2d_output,
+                                &conv2d_meta_output);
+
+          DataType dtype = DataTypeToEnum<T>::v();
+          TF_EXPECT_OK(
+              NodeDefBuilder("MklFusedBatchNorm", "_MklFusedBatchNormGradV3")
+                  .Input(FakeInput(dtype))
+                  .Input(FakeInput(dtype))
+                  .Input(FakeInput(DT_FLOAT))
+                  .Input(FakeInput(DT_FLOAT))
+                  .Input(FakeInput(DT_FLOAT))
+                  .Input(FakeInput(DT_FLOAT))
+                  .Input(FakeInput(DT_UINT8))
+                  .Input(FakeInput(DT_UINT8))
+                  .Input(FakeInput(DT_UINT8))
+                  .Input(FakeInput(DT_UINT8))
+                  .Input(FakeInput(DT_UINT8))
+                  .Input(FakeInput(DT_UINT8))
+                  .Attr("epsilon", epsilon)
+                  .Attr("is_training", true)
+                  .Attr("data_format", "NHWC")
+                  .Attr("_kernel", "MklLayoutDependentOp")
+                  .Finalize(node_def()));
+          TF_EXPECT_OK(InitOp());
+
+          AddInputFromArray<T>(y_backprop.shape(), y_backprop.flat<T>());
+          AddInputFromArray<T>(conv2d_output.shape(), conv2d_output.flat<T>());
+          AddInputFromArray<float>(scale.shape(), scale.flat<float>());
+          AddInputFromArray<float>(mean.shape(), mean.flat<float>());
+          AddInputFromArray<float>(variance.shape(), variance.flat<float>());
+          AddInputFromArray<float>(res_sp3.shape(), res_sp3.flat<float>());
+          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+          AddInputFromArray<uint8>(conv2d_meta_output.shape(),
+                                   conv2d_meta_output.flat<uint8>());
+          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+          TF_ASSERT_OK(RunOpKernel());
+
+          CommonTestUtilities<T> test_util;
+          test_util.PerformConversion(dtype, *GetOutput(0), *GetOutput(5),
```

Comment:
Isn't the mkl meta tensor at index 6 here?
```suggestion
          test_util.PerformConversion(dtype, *GetOutput(0), *GetOutput(6),
```
Same for `test_util_mean` and `test_util_var` (should indices 6 and 7 be 7 and 8 instead?)

Reply:
The indices in the code are correct as they are the output indices.

Reply:
Right. Thank you!